sos/cleaner/parsers/username_parser.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2020 Red Hat, Inc. Jake Hunsaker <jhunsake@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+
+from sos.cleaner.parsers import SoSCleanerParser
+from sos.cleaner.mappings.username_map import SoSUsernameMap
+
+
+class SoSUsernameParser(SoSCleanerParser):
+    """Parser for obfuscating usernames within an sosreport archive.
+
+    Note that this parser does not rely on regex matching directly, like most
+    other parsers do. Instead, usernames are discovered via scraping the
+    collected output of lastlog. As such, we do not discover new usernames
+    later on, and only usernames present in lastlog output will be obfuscated,
+    and those passed via the --usernames option if one is provided.
+    """
+
+    name = 'Username Parser'
+    map_file_key = 'username_map'
+    prep_map_file = 'sos_commands/login/lastlog_-u_1000-60000'
```

Comment:
Excellent choice to go with what I had implement in soscleaner to only obfuscate "internal" users created by useradd and such.

```
$ grep UID /etc/login.defs 
UID_MIN			 1000
UID_MAX			60000
```

Obfuscating public known users such as www-data, ubuntu, ... or else is unnecessary and based on my experience with soscleaner it will vastly contribute in having less false positive situations.

I'm definitely +1 on this approach.